src/app/pages/instruction-viewer/instruction-viewer.component.ts

```diff
@@ -528,52 +529,58 @@ export class InstructionViewerComponent implements OnInit, OnDestroy {
         const camera: OrthographicCamera = scene.userData["camera"];
         // get its position relative to the page's viewport
         const rect = element.getBoundingClientRect();
+        const rectParent = element.parentElement!.getBoundingClientRect();
         const canvasrect = this.canvas.getBoundingClientRect();
-        rect.x -= canvasrect.x;
-        // set the viewport
-        const width = rect.right - rect.left;
-        const height = rect.bottom - rect.top;
-        const left = rect.left;
-        const bottom = this.canvas.clientHeight - rect.bottom;
-        this.renderer.setViewport(left, bottom, width, height);
-        this.renderer.setScissor(left, bottom, width, height);
+        const top = rect.y - canvasrect.y;
+        const left = rect.x - canvasrect.x;
+        console.log(left, rect.height - top, rect.width, rect.height);
+        this.renderer.setSize(rectParent.width, rectParent.height);
+
+        this.renderer.setViewport(0, 0, canvasrect.width, canvasrect.height);
```

Comment:
here i handle the clearing of the rectangles but i do it in a bit off way

I first set the viewport to be the whole canvas, clear color and depth, and afterwards between the "boxes" i dont clear it a at all

so its cleared once per frame